### PR TITLE
Added support for reading carbon parameters in image driver

### LINKED ISSUE #390: Implement reading of carbon parameters in 5.0 image driver
Carbon parameters from the veg_lib file are not currently read in.

### PR DESCRIPTION
**Pull request by tbohn**
This is a branch off of PR #389.

Closes #390.

### CODE REVIEW

File: vic/drivers/classic/src/read_veglib.c [hunk 1]
```diff
@@ -163,10 +163,10 @@ read_veglib(FILE   *veglib,
             /* Carbon-cycling parameters */
             if (options.VEGLIB_PHOTO) {
                 fscanf(veglib, "%s", tmpstr); /* photosynthetic pathway */
-                if (!strcmp(tmpstr, "C3")) {
+                if (!strcmp(tmpstr, "0") || !strcmp(tmpstr, "C3")) {
                     temp[i].Ctype = PHOTO_C3;
                 }
-                else if (!strcmp(tmpstr, "C4")) {
+                else if (!strcmp(tmpstr, "1") || !strcmp(tmpstr, "C4")) {
```
Comment (jhamman) [inline vic/drivers/classic/src/read_veglib.c]: Since you're deprecating `0` and `1`, can you print a warning statement if those options are used?

Comment (tbohn) [inline vic/drivers/classic/src/read_veglib.c]: OK, will do.  Except that it's the strings "C3" and "C4" that I'm deprecating.  And the reason for that is that I had trouble figuring out how to get them to work correctly in tonic - I tried for hours - and finally relented and made them ints.  So, tonic can read either of them (string or int), but will convert to the 0/1 values for writing to netcdf.  So, VIC is reading them as ints in image mode.

Comment (jhamman) [inline vic/drivers/classic/src/read_veglib.c]: Why don't we figure out how to fix your `tonic` code?  If you can show me what you weren't able to get to work, I'm sure I can help you get it to work. No need to change VIC.

Comment (jhamman) [inline vic/drivers/classic/src/read_veglib.c]: okay, after further thought and an offline conversation with @tbohn - I think this is fine for now.

**Comment by jhamman** [comment]
merging now...

File: vic/drivers/classic/src/read_veglib.c (hunk 1 above)
Comment (tbohn) [inline vic/drivers/classic/src/read_veglib.c]: :-)

**Pull request merged**